lock-api/src/main/java/com/palantir/lock/client/UnreliableTimeLockService.java

```diff
@@ -31,30 +32,52 @@
 import com.palantir.logsafe.SafeArg;
 import com.palantir.logsafe.logger.SafeLogger;
 import com.palantir.logsafe.logger.SafeLoggerFactory;
+import com.palantir.timestamp.TimestampManagementService;
 import com.palantir.timestamp.TimestampRange;
+import java.security.SecureRandom;
 import java.util.List;
 import java.util.Set;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
 /**
  * A version of timelock service that has a chance to randomly lose locks during refresh or immediately after
  * acquiring. This is useful for testing the behavior of clients when locks are lost.
  */
 public final class UnreliableTimeLockService implements TimelockService {
```

Comment:
Discussed offline: It might be better for maintainability to separate the locking logic into its separate service class and use that here: that can be complicated given locking / buggifying are intertwined, might not be worth it.

The only thing I'd want as a minimum is to rename the lock or comment it heavily or rename usage methods (or heck even separate it out) and make sure the write lock is only ever taken for ff and nothing else (i.e. if you spin a separate class, you could call the run method, runForFF or stg).